### PR TITLE
BFD-354: Document HTTP request headers (for audit and optional data)

### PR DESCRIPTION
**Pull request by karlmdavis**
### Change Details

Add docs on HTTP request headers.

Covers the headers used for audit purposes and those that toggle optional data in the results.

Please note that I also included a bunch of formatting and whitespace changes here in the `README.md`, to:

1. Resolve Git issues with the file.
2. Ensure that it made sense, from an info architecture angle.

(You'll want to either review the commits individually or suppress whitespace changes when reviewing.)

https://jira.cms.gov/browse/BFD-354

### Acceptance Validation

1. Is the documentation clear?
2. Is the documentation correct?
3. Is the documentation complete (or complete enough for now)?

### Feedback Requested

(See Acceptance Criteria, above.)

### External References

N/A

### Security Implications

No security implications noted here:

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Review by jzulim — APPROVED** [review]
Documentation feels clear, updates to the main README make sense. I agree that we should pull out the local dev into it's own markdown and reference here like you have the other docs. Header documentation / requirements for sync and async will help us from an operational and audit perspective. Thumbs up from me.

File: docs/request-audit-headers.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+* `BlueButton-OriginalQueryTimestamp`: an ISO-8601 UTC timestamp representing (roughly) when the original request reached the frontend system.
+* `BlueButton-DeveloperId`: the unique ID in the frontend system for the developer/vendor of the third-party application.
+* `BlueButton-Developer`: the human-readable name in the frontend system for the developer/vendor of the third-party application.
+* `BlueButton-ApplicationId`: the unique ID in the frontend system for the application/client that will receieve the data.
+* `BlueButton-Application`: the human-readable name in the frontend system for the application/client that will receive the data.
+* `BlueButton-UserId`: the unique ID in the frontend system for the user (e.g. beneficiary) that data is being requested on behalf of.
+* `BlueButton-User`: the human-readable login ID (e.g. email address) in the frontend system for the user (e.g. beneficiary) that the data is being requested on behalf of.
+* `BlueButton-BeneficiaryId`: the unique beneificary ID (e.g. `Patient.id`) from BFD for the user the beneficiary whose data is being requested.
+
+## Asynchronous Requests
```
Comment (jzulim) [inline docs/request-audit-headers.md:29]: @keithdadkins - This should do the trick for using header to aggregate bulk metrics. It'll also allow us to further breakdown these bulk partner requests by the requesting party (ACO, Provider, etc.)

File: docs/request-options.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+This document details the request options that can be used when calling BFD.
+Future versions of BFD may apply some of these options automatically,
+  based on other request details (e.g. authentication/authorization).
+
+## Optional Data
+
+Some data fields are optional; they're only included when the request is configured to do so.
+
+* HTTP Header: `IncludeIdentifiers`
```
Comment (jzulim) [inline docs/request-options.md:11]: I did not know how this behavior worked, this documentation helped a lot. Thx!

**Review by dshekhar18 — APPROVED** [review]
Good materials for SDD. Thanks @karlmdavis

**Pull request merged**